svsim/src/main/scala/verilator/Backend.scala

```diff
@@ -8,9 +8,55 @@ import scala.collection.mutable
 
 object Backend {
   object CompilationSettings {
-    sealed trait TraceStyle
-    object TraceStyle {
-      case class Vcd(traceUnderscore: Boolean = false, filename: String = "") extends TraceStyle
+
+    sealed trait TraceKind {
+      private[Backend] def toCompileFlags: Seq[String]
+    }
+
+    object TraceKind {
+
```

Comment:
This is a mega-nit, but I prefer enums like this:

```suggestion
    object TraceKind {
      sealed trait Type {
        private[Backend] def toCompileFlags: Seq[String]
      }

```